### PR TITLE
User DateExport

### LINKED ISSUE #380: Add dataExport api call to core
This is missing from core
https://github.com/Stremio/stremio-api/blob/752d336f2d52740312686c80867b5f7b1281861b/docs/dataExport.md

### PR DESCRIPTION
**Pull request by elpiel**
Signed-off-by: Lachezar Lechev <lachezar@ambire.com>

### CODE REVIEW

File: src/models/data_export.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#[derive(Serialize, Default, Clone)]
+#[cfg_attr(debug_assertions, derive(Debug))]
+#[serde(rename_all = "camelCase")]
+pub struct DataExport {
+    /// This is the Loading result of the User data export request.
+    pub export_url: Option<(AuthKey, Loadable<Url, CtxError>)>,
+}
+
+impl DataExport {
+    pub fn new() -> Self {
```
Comment (nklhtv) [inline src/models/data_export.rs]: use DataExport::default()
we use ::new() only if it returns some effects

File: src/unit_tests/data_export.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        Request {
+            url: "https://api.strem.io/api/dataExport".to_owned(),
+            method: "POST".to_owned(),
+            headers: Default::default(),
+            body: r#"{"type":"DataExport","authKey":"user_key"}"#.to_owned(),
+        }
+    );
+}
+
+// https://www.strem.io/data-export/{}/export.json
```
Comment (nklhtv) [inline src/unit_tests/data_export.rs]: that comment should be deleted

File: src/unit_tests/data_export.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+#[test]
+fn data_export_with_user() {
+    let _env_mutex = TestEnv::reset();
+    *FETCH_HANDLER.write().unwrap() = Box::new(data_export_fetch_handler);
+    let mut ctx = Ctx::default();
+    ctx.profile.auth = Some(Auth {
+        key: AuthKey("user_key".into()),
+        user: User {
```
Comment (nklhtv) [inline src/unit_tests/data_export.rs]: you can use User::default() for the tests

**Review by nklhtv — CHANGES_REQUESTED** [review]

File: src/unit_tests/data_export.rs (hunk 3 above)
Comment (elpiel) [reply src/unit_tests/data_export.rs]: Fixed.

File: src/unit_tests/data_export.rs (hunk 2 above)
Comment (elpiel) [reply src/unit_tests/data_export.rs]: fixed.

File: src/models/data_export.rs (hunk 1 above)
Comment (elpiel) [reply src/models/data_export.rs]: :+1: fixed.

**Review by nklhtv — APPROVED** [review]

**Pull request merged**